src/item_factory.cpp

```diff
@@ -2003,12 +2025,26 @@ void armor_portion_data::deserialize( const JsonObject &jo )
     } else {
         optional( jo, false, "encumbrance", encumber, 0 );
     }
-    optional( jo, false, "material_thickness", thickness, 0.0f );
+    optional( jo, false, "material_thickness", avg_thickness, 0.0f );
     optional( jo, false, "environmental_protection", env_resist, 0 );
     optional( jo, false, "environmental_protection_with_filter", env_resist_w_filter, 0 );
 
     // TODO: Make mandatory - once we remove the old loading below
-    optional( jo, false, "material", materials );
+    if( jo.has_member( "material" ) ) {
+        if( jo.has_array( "material" ) && jo.get_array( "material" ).test_object() ) {
+            mandatory( jo, false, "material", materials );
+        } else {
+            // Old style material definition ( ex: "material": [ "cotton", "plastic" ] )
+            // TODO: Depricate and remove
+            for( const std::string &mat : jo.get_tags( "material" ) ) {
+                part_material m;
+                m.id = material_id( mat );
+                m.portion = 1;
+                m.thickness = 0.0f;
+                materials.emplace_back( m );
```

Comment:
You can define a parameterized constructor for `part_material` so you can construct it in-place like
```cpp
materials.emplace_back( material_id( mat ), 1, 0.0f );
```

Reply:
Thanks! I changed it to use the constructor.